### PR TITLE
Minimal tests with inventory v4

### PR DESCRIPTION
**Pull request by jotak**
First step to validate the api, but for the time being it doesn't use cassettes, security context etc.
Hard-coded server url

### CODE REVIEW

**Comment by jotak** [comment]
Note: the tests are passing for me, but I need to start my own hawkular server build aside (no hawkular-services docker image built yet)

**Comment by jotak** [comment]
PS: I just built a docker image for hawkular-services with new inventory: `jotak/hawkular-services:0.40.0.Final-SNAPSHOT` ... cc @josejulio

File: lib/hawkular/inventory/inventory_api_v4.rb [hunk 1]
```diff
@@ -37,29 +37,29 @@ def self.create(hash)
     # Get single resource by id
     # @return Resource the resource
     def resource(id)
-      hash = http_get("/resources/#{id}")
+      hash = http_get("/resources/#{URI.escape(id)}")
```
Comment (cfcosta) [inline lib/hawkular/inventory/inventory_api_v4.rb:40]: I think http_get should escape the URI directly, instead of adding it on each call.

**Review by cfcosta — CHANGES_REQUESTED** [review]
Looks good, but I think changing the URI escaping to be more consistent is necessary.

File: lib/hawkular/inventory/inventory_api_v4.rb (hunk 1 above)
Comment (jotak) [reply lib/hawkular/inventory/inventory_api_v4.rb:40]: that would be a big refactor, as `http_get` is used in other subclasses and some of them perform a non-standard escaping that couldn't be combined with URI.escape (see https://github.com/hawkular/hawkular-client-ruby/blob/master/lib/hawkular/client_utils.rb#L20 )

File: lib/hawkular/inventory/inventory_api_v4.rb [hunk 2]
```diff
@@ -37,29 +37,29 @@ def self.create(hash)
...
-      hash = http_get("/resources/#{id}/tree")
+      hash = http_get("/resources/#{URI.escape(id)}/tree")
       Resource.new(hash)
     end
 
     # List root resources
     # @return [Array<Resource>] List of resources
     def root_resources
-      # FIXME: pagination
+      # FIXME: pagination => lazy-loaded list with ruby?
```
Comment (josejulio) [inline lib/hawkular/inventory/inventory_api_v4.rb:54]: +1

File: lib/hawkular/inventory/inventory_api_v4.rb (hunk 1 above)
Comment (josejulio) [reply lib/hawkular/inventory/inventory_api_v4.rb:40]: Agree, lets review that on a separate PR

**Pull request merged**